openlibrary/components/AuthorIdentifiers.vue

```diff
@@ -0,0 +1,133 @@
+<template>
+  <table>
+    <tr>
+      <th>
+        <select class="form-control" v-model="selectedIdentifier" name="name">
+          <option disabled value="">Select one</option>
+          <option v-for="idConfig in identifierConfigsByKey" :key="idConfig.name" :value="idConfig.name">
+            {{idConfig.label}}
+          </option>
+        </select>
+      </th>
+      <th>
+        <input class="form-control" type="text" name="value" id="id-value" v-model="inputValue" @keyup.enter=setIdentifier>
+      </th>
+      <th>
+        <button class="form-control" type="button" name="set" :disabled="!setButtonEnabled" @click=setIdentifier>Set</button>
+      </th>
+    </tr>
+    <tr :key="identifier.name" v-for="identifier in assignedIdentifiersWithConfigs">
+      <td>{{ identifier.label }}</td>
+      <td>{{ identifier.value }}</td>
+      <td>
+        <button class="form-control" type="button" @click="removeIdentifier(identifier.name)">Remove</button>
+      </td>
+    </tr>
+  </table>
+</template>
+
+<script>
+export default {
+    // Props are for external options; if a subelement of this is modified,
+    // the view automatically re-renders
+    props: {
+        /** The list of ids currently associated with the entity in the database in string form */
+        assigned_ids_string: {
+            type: String,
+            //default: () =>  "{'wikidata': 'Q10000'}"
+        },
+        /** everything from https://openlibrary.org/config/author */
+        author_config: {
+            type: Object
+        },
+        /** see createHiddenInputs function for usage */
+        output_selector: {
+            type: String
+        }
+    },
+
+    // Data is for internal stuff. This needs to be a function so that we get
+    // a fresh object every time this is initialized.
+    data: () => {
+        return {
+            selectedIdentifier: '', // Which identifier is selected in dropdown
+            inputValue: '', // What user put into input
+            assignedIdentifiers: {}, // IDs assigned to the entity
+        }
+    },
+
+    computed: {
+        /** A list of assigned identifiers, merged with their respective configs */
+        assignedIdentifiersWithConfigs: function(){
```

Comment:
I think this is what's making these a little confusing. The return type of this is kind of complicated, and difficult to discern. Let's flatten the logic, and dissolve `assignedIdentifiersWithConfigs` and reference `identifierConfigsByKey` directly instead.